nextjs-13/src/lib/handler/base.handler.ts

```diff
@@ -0,0 +1,37 @@
+import { User } from '@prisma/client';
+import { NextResponse, NextRequest } from 'next/server';
+import { getToken } from 'next-auth/jwt';
+
+import { findUserById } from 'repositories/user.repository';
+
+export default async function baseHandler(
+  req: NextRequest,
+  callback: (currentUser: User, body: unknown) => Promise<NextResponse>
+) {
+  try {
+    const token = await getToken({ req });
+
+    if (token && token.userId) {
+      const currentUser = await findUserById(token.userId as string);
+
+      if (currentUser) {
+        let body = {};
+
+        try {
+          body = await req.json();
+        } catch (err) {}
```

Comment:
Not sure why this catch is empty 😅

Reply:
It's because that, for `POST` requests, the `req` will have the body. But for `GET` requests, the requests not have body, so when it executes `req.json()`, it doesn't just return an empty hash `{}`, but it raises error. So this line just means that if we can extract the body, then extract(`POST`). Else(`GET`), don't do anything, and don't raise error too please 😄 .

Reply:
```
 if (req.method === 'POST') {
    // Process a POST request
  }
```
maybe you can try this instead, not sure it is work in App route [Reference](https://nextjs.org/docs/pages/building-your-application/routing/api-routes).

Reply:
Yeah I have thought of that too, but if we want to use if else statements instead, we then also have to use if else for `PUT`, `PATCH`, `DELETE`, ... The try-catch seems lazy but I think it may help us make the code less verbose a bit, no? 👀

Reply:
Ya in this case try-catch make the code less verbose, maybe just use try-catch should be fine here.

Reply:
I am still. not convinced 😅 Prefer early returns over long if/else blocks + rely on the catch block to handle callback errors:

```javascript
const token = await getToken({ req });
const currentUser = await findUserById(token?.userId as string);

if (!currentUser) {
  // raise auth error
}

return await callback(currentUser, req.json());
```

Reply:
Yeah I think early returns are better too.

So to keep this simple, I will let the APIs handle the `req.json()` part themself for now, so now the code is simplified into [this](https://github.com/nimblehq/js-framework-benchmark/pull/96/files#diff-10b91e667909ffe6c8b2d34d735e0fa46be166ba37cf1b5ec24f7ea6b54658bbR12-R22).